test/server/active_tcp_listener_test.cc

```diff
@@ -0,0 +1,181 @@
+#include <memory>
+
+#include "envoy/network/filter.h"
+#include "envoy/network/listener.h"
+#include "envoy/stats/scope.h"
+
+#include "common/network/address_impl.h"
+#include "common/network/connection_balancer_impl.h"
+#include "common/network/raw_buffer_socket.h"
+#include "common/network/utility.h"
+
+#include "server/active_tcp_listener.h"
+
+#include "test/mocks/api/mocks.h"
+#include "test/mocks/common.h"
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/network_utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::Invoke;
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Server {
+namespace {
+
+class MockTcpConnectionHandler : public Network::TcpConnectionHandler,
+                                 public Network::MockConnectionHandler {
+public:
+  MOCK_METHOD(Event::Dispatcher&, dispatcher, ());
+  MOCK_METHOD(Network::BalancedConnectionHandlerOptRef, getBalancedHandlerByTag,
+              (uint64_t listener_tag));
+  MOCK_METHOD(Network::BalancedConnectionHandlerOptRef, getBalancedHandlerByAddress,
+              (const Network::Address::Instance& address));
+};
+class ActiveTcpListenerTest : public testing::Test, protected Logger::Loggable<Logger::Id::main> {
+public:
+  ActiveTcpListenerTest() {
+    EXPECT_CALL(conn_handler_, dispatcher()).WillRepeatedly(ReturnRef(dispatcher_));
+    EXPECT_CALL(conn_handler_, numConnections()).Times(testing::AnyNumber());
```

Comment:
Is it necessary ro specify the number of calls if it's `AnyNumber`? Or does that imply greater than zero?

Reply:
My hunch is that AnyNumber is 0 or more. 
This PR has only 1 test case for now. I plan to migrate more test cases more conn_handler_impl_test here. Some test cases mock numConnections() and leave the numConnections() calls 0 times in this EXPECT_CALL.